### PR TITLE
Joss Review changes

### PR DESCRIPTION
**Pull request by AlexBuccheri**
Paper and source code changes in response to [JOSS paper submission](https://github.com/openjournals/joss-reviews/issues/5148)

**Changes**:

- Remove dangling issue number in README.md
- Auto-format README.md
- Add .idea and .vscode folders, providing more IDE support
- Remove ambiguous, inaccurate line referring to relative NOMAD package size
- Add pylint and yapf configurations, which were missing
- Add additional references ML data fidelity requirements, and LAPW basis set
- Reference usage of excitingtools in workflows, in the README.md

@speckhard and @FabiPi3 so they're aware of the status

- [x] Clean up commits prior to merging
   -  Commits were clean, so I've not needed to rebase

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,17 +111,15 @@ pip install pdoc
 pdoc -o documentation -d restructuredtext --math excitingtools/
 ```
 
-- [ ] TODO(Alex) Issue 57. Set up generation of documentation from docstrings, with Sphinx
```
Comment (AlexBuccheri) [inline README.md:105]: Removed line in response to JOSS review

File: paper/paper.md [hunk 2]
```diff
@@ -120,9 +120,7 @@ integration/deployment model, such that new features and updates are delivered s
     downloading the [NOMAD parsers](https://github.com/nomad-coe/nomad-parser-exciting) which return custom objects, 
     containing copious metadata. This is unnecessary for <span style="font-family:american typewriter; font-size:1em;">**exciting**</span>
     users, and prevents straightforward numerical comparison of parsed results.
-    - _excitingtools_ (plus dependencies) is significantly more lightweight than the NOMAD package, approximately two
```
Comment (AlexBuccheri) [inline paper/paper.md:123]: Remove inaccurate statement

**Review by speckhard — APPROVED** [review]
Looks great. Thanks for making these changes.

**Review by hmacdope — APPROVED** [review]
Looks good to me!

**Pull request merged**